solvers/celer.py

```diff
@@ -20,29 +23,49 @@ class Solver(BaseSolver):
     ]
 
     def set_objective(self, X, y, lambdas, fit_intercept):
+        # celer/sklearn way of handling intercept: center X and y for dense
+        if fit_intercept:
+            X, y, X_offset, y_offset, _ = _preprocess_data(
+                X, y, fit_intercept, return_mean=True, copy=True,
+            )
+            self.X_offset = X_offset
+            self.y_offset = y_offset
+
         self.X, self.y = X, y
         self.lambdas = lambdas
         self.fit_intercept = fit_intercept
 
-    def skip(self, X, y, lambdas, fit_intercept):
-        if fit_intercept:
-            return True, f"{self.name} does not handle fit_intercept"
-
-        return False, None
-
     def run(self, tol):
         warnings.filterwarnings("ignore", category=ConvergenceWarning)
 
-        _, self.coefs, _ = celer_path(
-            self.X,
-            self.y,
-            pb="lasso",
-            alphas=self.lambdas / len(self.y),
-            prune=1,
-            tol=tol,
-            max_iter=1_000,
-            max_epochs=100_000,
-        )
+        if self.fit_intercept and sparse.issparse(self.X):
```

Comment:
I would just define X_offset and X_scale inside the if, 
then define them as None in the other case, and do a single call to celer_path to factorize the code

Reply:
Ok, will do it